packages/vitest/src/node/cli-wrapper.ts

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable no-console */
+/**
+ * Wrapper of the CLI with child process to manage segfaults and retries.
+ */
+import { fileURLToPath } from 'url'
+import c from 'picocolors'
+import { execa } from 'execa'
+
+const ENTRY = new URL('./cli.mjs', import.meta.url)
+
+interface ErrorDef {
+  trigger: string
+  url: string
+}
+
+// Node errors seen in Vitest (vitejs/vite#9492)
+const ERRORS: ErrorDef[] = [
+  {
+    trigger: 'Check failed: result.second.',
+    url: 'https://github.com/nodejs/node/issues/43617',
+  },
+  {
+    trigger: 'FATAL ERROR: v8::FromJust Maybe value is Nothing.',
+    url: 'https://github.com/vitest-dev/vitest/issues/1191',
+  },
+  {
+    trigger: 'FATAL ERROR: v8::ToLocalChecked Empty MaybeLocal.',
+    url: 'https://github.com/nodejs/node/issues/42407',
+  },
+]
+
+async function main() {
+  // default exit code = 100, as in retries were exhausted
+  const exitCode = 100
+  let retries = 0
+  const args = process.argv.slice(2)
+
+  if (process.env.VITEST_SEGFAULT_RETRY) {
+    retries = +process.env.VITEST_SEGFAULT_RETRY
+  }
+  else {
+    for (let i = 0; i < args.length; i++) {
+      if (args[i].startsWith('--segfault-retry=')) {
+        retries = +args[i].split('=')[1]
+        break
+      }
+      else if (args[i] === '--segfault-retry' && args[i + 1]?.match(/^\d+$/)) {
+        retries = +args[i + 1]
+        break
+      }
+    }
+  }
+
+  retries = Math.max(1, retries || 1)
+
+  for (let i = 1; i <= retries; i++) {
+    if (i !== 1)
+      console.log(`${c.inverse(c.bold(c.magenta(' Retrying ')))} vitest ${args.join(' ')} ${c.gray(`(${i} of ${retries})`)}`)
+    await start(args)
+    if (i === 1 && retries === 1) {
+      console.log(c.yellow(`It seems to be an upstream bug of Node.js. To improve the test stability,
+you could pass ${c.bold(c.green('--segfault-retry=3'))} or set env ${c.bold(c.green('VITEST_SEGFAULT_RETRY=3'))} to
+have Vitest auto retries on flaky segfaults.\n`))
+    }
+  }
+  process.exit(exitCode)
+}
+
+main()
+
+async function start(args: string[]) {
+  const child = execa('node', [fileURLToPath(ENTRY), ...args], {
```

Comment:
Doesn't it hurt performance for "--segfault-retry=1" that we always starts a child process? Shouldn't we just import `./cli.mjs` in this case?

Reply:
Hm, but we can't show useful error message that way 🤔 

I guess it's ok for now.

Reply:
I didn't count how many times we invoke `vitest` in our test, but at the high level, it doesn't make CI time increase. Using this would allow us to provide information about the segfault reasoning and issue link.